props/blaster.h

```diff
@@ -260,9 +233,92 @@ class Blaster : public PROP_INHERIT_PREFIX PropBase {
     }
   }
 
+  RefPtr<BufferedWavPlayer> auto_player_;
+
+  uint32_t auto_time_;
+  virtual void Fire() {
+    CheckEmpty();
+    if (empty_) return;
+#ifdef ENABLE_MOTION
+#ifdef BLASTER_JAM_PERCENTAGE
+    // If we're already jammed then we don't need to recheck. If we're not jammed then check if we just jammed.
+    is_jammed_ = is_jammed_ ? true : CheckJam(BLASTER_JAM_PERCENTAGE);
+    if (is_jammed_) {
+      SaberBase::DoEffect(EFFECT_JAM, 0);
+      return;
+    }
+#endif
+#endif
+    if (blaster_mode == MODE_AUTO) {
+      // Set up the autofire pairing if the font suits it.
+      SelectAutoFirePair();
+      auto_firing_ = true;
+      auto_time_ = millis();
+      SaberBase::SetLockup(LOCKUP_AUTOFIRE);
+      SaberBase::DoBeginLockup();
+      STDOUT << "***************************** OK Going to GetNewestAutoWav() with &SFX_auto in tow\n";
+      GetNewestAutoWav(&SFX_auto);
+      STDOUT << "***************************** Back here now and doing autofire like I should. \n";
+
+    } else {
+      if (blaster_mode == MODE_STUN) {
+        SaberBase::DoEffect(EFFECT_STUN, 0);
+        STDOUT << "******** STUN - Remaining shots = " << GetBulletCount() << "\n";
+      } else {
+        SaberBase::DoEffect(EFFECT_FIRE, 0);
+        STDOUT << "******** FIRE - Remaining shots = " << GetBulletCount() << "\n";
+      }
+      shots_fired_++;
+    }
+  }
+
+  RefPtr<BufferedWavPlayer> GetNewestAutoWav(Effect* a) {
+    // auto_player_ = GetWavPlayerPlaying(&SFX_auto);
+    float lowest_progress = 360000;
+    uint32_t player_to_choose = 0;
+    // check wavplayers playing auto.wav, choose newest to use 
+    for (size_t i = 0; i < NELEM(wav_players); i++) {
+    STDOUT << "***************************** Now inside GetNewestAutoWav for-loop.   i = " << i << "\n";
+// about check wavplayers for auto.wav, and is not referenced
+        if (wav_players[i].isPlaying() &&
+        wav_players[i].current_file_id().GetEffect() == a &&
+        wav_players[i].refs() == 0) {
+        STDOUT << "***************************** This if-clause is not turning up true EVER. trying to do stuff \n";
+        float pos = wav_players[i].pos();
+        if (pos < lowest_progress) {
+          STDOUT << "***************************** pos < lowest_progress = " << pos << "\n";
+          lowest_progress = pos;
+          player_to_choose = i;
+          STDOUT << "***************************** player_to_choose = " << player_to_choose << "\n";
+        }    
+      }
+    }
+    STDOUT << "***************************** for-loop done, chosen player = " << player_to_choose << " \n";
+    auto_player_ = RefPtr<BufferedWavPlayer>(wav_players + player_to_choose);
+    return auto_player_;
+  }
+
+
   void Loop() override {
     PropBase::Loop();
     PollNextAction();
+    if (auto_firing_) {
```

Comment:
if (A) {
  if (B) {
  }
}

should almost always be written as:
  if (A && B) {
  }